common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -750,6 +753,81 @@ public AccountDeletionRecord removeAccount(@Nullable final String environment,
         );
     }
 
+    /**
+     * Tests if a clientId is 'known' to the cache. A clientId is known if a token has been
+     * previously saved to the cache with it.
+     *
+     * @param clientId The clientId whose known status should be queried.
+     * @return True if this clientId is known. False otherwise.
+     */
+    public boolean isClientIdKnownToCache(@NonNull final String clientId) {
+        return getAllClientIds().contains(clientId);
```

Comment:
When will the known clientIds set get updated? By saving a new token?

Reply:
Yes